cms/admin/forms.py

```diff
@@ -926,49 +921,42 @@ def __init__(self, *args, **kwargs):
         self._site = kwargs.pop("site", Site.objects.get_current())
         super().__init__(*args, **kwargs)
         self.fields["target"].queryset = Page.objects.filter(
-            node__site=self._site,
+            site=self._site,
             is_page_type=self.page.is_page_type,
         )
 
-    def get_root_nodes(self):
```

Comment:
The modified pull request now includes:

> A `node` property for Page that returns the page itself. Thereby, custom apps that access page.node.something will not break.

**done**

> Remove `node__` prefixes from `filter`, `order_by`, `select_related` in `PageQuerySet`

**done**

> Rename `page.parent` to `page.page_parent` to be compatible with django CMS 4.1 and earlier

`page.page_parent` now accesses `page.parent` and issues a deprecation warning. By convenience `parent` is the term used in all trees. This was already the case for `TreeNode` and is also used in [django-tree-queries](https://django-tree-queries.readthedocs.io/en/latest/#features-and-limitations).

> Use the RemovedInDjangoCMS43 warning message to mark compatibility code that will need to be removed in the future.

**done**


I also removed all the legacy code including model `TreeNode` itself.

Reply:
> On the smaller things, I've noticed that you use both single and double quotes (' and "). I'd suggest being consistent and opting for double quotes

Django-CMS is not very consistent on this and Django only since version 4.2.

In my projects I use this rule: If it's intended for humans, I use `"…"`-quotes. If it's an internal string (key, fieldname, attribute, etc.), I use `'…'`-quotes. This idea was propagated by Douglas Crockford many years ago.

Reply:
> I would like to opt for better readability here and avoid the walrus operator at least in a nested statement (if not at all).

In my opinion less lines of code are easier to read. Therefore Python's decision to add the walrus operator to the language makes a lot of sense. I now removed it in `not(…)`-statements, although I would prefer to keep it.